pkg/internal/cnsoperator/cnsfilevolumeclient/cnsfilevolumeclient.go

```diff
@@ -91,6 +94,50 @@ func GetFileVolumeClientInstance(ctx context.Context) (FileVolumeClient, error)
 	return fileVolumeClientInstance, nil
 }
 
+// GetClientVmsFromIPList returns the list of client vms associated with a given External IP address and CnsFileVolumeClient instance
+// Returns an error if any operations fails.
+func (f *fileVolumeClient) GetClientVmsFromIPList(ctx context.Context, fileVolumeName string, clientVMIP string) ([]string, error) {
+	log := logger.GetLogger(ctx)
+
+	log.Infof("Fetching client VMs list from cnsfilevolumeclient %s for IP address %s", fileVolumeName, clientVMIP)
+	actual, _ := f.volumeLock.LoadOrStore(fileVolumeName, &sync.Mutex{})
+	instanceLock, ok := actual.(*sync.Mutex)
+	if !ok {
+		return nil, fmt.Errorf("failed to cast lock for cnsfilevolumeclient instance: %s", fileVolumeName)
+	}
+	instanceLock.Lock()
+	defer instanceLock.Unlock()
+
+	instance := &v1alpha1.CnsFileVolumeClient{}
+	instanceNamespace, instanceName, err := cache.SplitMetaNamespaceKey(fileVolumeName)
+	if err != nil {
+		log.Errorf("failed to split key %s with error: %+v", fileVolumeName, err)
+		return []string{}, err
+	}
+	instanceKey := types.NamespacedName{
+		Namespace: instanceNamespace,
+		Name:      instanceName,
+	}
+	err = f.client.Get(ctx, instanceKey, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// If the get() on the instance fails, then we return empty list.
+			log.Infof("Cnsfilevolumeclient instance %s not found: %+v. Returning empty list", fileVolumeName, err)
```

Comment:
I dont think we need to also print `err` in this log message. It will probably clutter the log

Reply:
Done